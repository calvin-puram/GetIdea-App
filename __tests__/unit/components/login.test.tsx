import React from 'react';
import { axe } from 'jest-axe';
import { render, fireEvent } from '@testing-library/react';
import ReactDom from 'react-dom';

import Login from '@pages/login';

describe('The <Login /> Component', () => {
  const setupLogin = () => render(<Login />);

  const label = {
    email: 'Email',
    password: 'Password',
  };

  it('Should render the email field the user will see', () => {
    const { asFragment } = setupLogin();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the password field the user will see', () => {
    const { asFragment } = setupLogin();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should make the email field accessible by setting the id and htmlFor attributes on label and input', () => {
    const { getByLabelText } = setupLogin();
    expect(getByLabelText(label.email)).toBeInTheDocument();
  });

  it('Should make the password field accessible by setting the id and htmlFor attributes on label and input', () => {
    const { getByLabelText } = setupLogin();
    expect(getByLabelText(label.password)).toBeInTheDocument();
  });

  it('Should not fail any accessibility test', async () => {
    const { container } = setupLogin();
    expect(await axe(container)).toHaveNoViolations();
  });
});
