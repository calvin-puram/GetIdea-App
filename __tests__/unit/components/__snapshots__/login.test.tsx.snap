// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Login /> Component Should render the email field the user will see 1`] = `
<DocumentFragment>
  <section
    class="loginWrapper"
  >
    <div
      class="row"
    >
      <div
        class="loginSeparator1 col"
      >
        <a
          class="back__redirect nav-link"
          href="/"
        >
          <svg
            class="backButton"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
            />
          </svg>
           
          <span>
            Back
          </span>
        </a>
        <div
          style="max-width: 100%; width: 150px;"
        >
          <div
            style="position: relative; padding-bottom: 43.333333333333336%;"
          >
            <img
              alt="login"
              class="__lazy"
              data-src="/_next/image?url=%2Fimg%2Floginmatrix.png&w=320&q=75"
              data-srcset="/_next/image?url=%2Fimg%2Floginmatrix.png&w=320&q=75 320w"
              style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
          </div>
        </div>
        <div
          class="loginFormSubWrapper"
        >
          <div
            class="logo__section"
          >
            <div
              class="logo__title"
            >
              GetIdea
            </div>
            <div
              style="max-width: 100%; width: 35px;"
            >
              <div
                style="position: relative; padding-bottom: 108.57142857142857%;"
              >
                <img
                  alt="logo"
                  class="__lazy"
                  data-src="/_next/image?url=%2Fimg%2Flogo.png&w=320&q=75"
                  data-srcset="/_next/image?url=%2Fimg%2Flogo.png&w=320&q=75 320w"
                  style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
              </div>
            </div>
          </div>
          <p
            class="loginPara"
          >
            Sign in to your account to continue
          </p>
          <form
            action="#"
            class="loginForm"
          >
            <div
              class="loginForm__seperator"
            >
              <label
                class="label"
                for="email"
              >
                Email
              </label>
              <input
                class="loginForm__input"
                id="email"
                name="email"
                type="text"
                value=""
              />
            </div>
            <div
              class="loginForm__seperator"
            >
              <label
                class="label"
                for="password"
              >
                Password
              </label>
              <input
                class="loginForm__input"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="loginForm__authRedirect"
            >
              <a
                class="loginForm__redirect nav-link"
                href="/forgot-password"
              >
                Forgot Password?
              </a>
              <a
                class="loginForm__redirect nav-link"
                href="/register"
              >
                Donâ€™t have an account 
                <span>
                  Register
                </span>
                ?
              </a>
            </div>
            <div
              class="loginForm__seperator"
            >
              <button
                class="loginForm__submit"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
        <div
          style="max-width: 100%; width: 647px;"
        >
          <div
            style="position: relative; padding-bottom: 21.020092735703248%;"
          >
            <img
              alt="login squares"
              class="loginSeparator1Img __lazy"
              data-src="/_next/image?url=%2Fimg%2Floginranbows.png&w=768&q=75"
              data-srcset="/_next/image?url=%2Fimg%2Floginranbows.png&w=320&q=75 320w, /_next/image?url=%2Fimg%2Floginranbows.png&w=420&q=75 420w, /_next/image?url=%2Fimg%2Floginranbows.png&w=768&q=75 768w"
              style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
          </div>
        </div>
      </div>
      <div
        class="loginSeparator2 col"
      >
        <div
          class="loginSideWrapper"
        >
          <div
            style="max-width: 100%; width: 50px;"
          >
            <div
              style="position: relative; padding-bottom: 90%;"
            >
              <img
                alt="login icon"
                class="__lazy"
                data-src="/_next/image?url=%2Fimg%2Floginsideicon.png&w=320&q=75"
                data-srcset="/_next/image?url=%2Fimg%2Floginsideicon.png&w=320&q=75 320w"
                style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
            </div>
          </div>
          <p
            class="loginSidePara"
          >
            Our support team are always ready 24/7 to help you start and grow your business
          </p>
          <div
            style="max-width: 100%; width: 50px;"
          >
            <div
              style="position: relative; padding-bottom: 50%;"
            >
              <img
                alt="login icon"
                class="__lazy"
                data-src="/_next/image?url=%2Fimg%2Floginsideicon2.png&w=320&q=75"
                data-srcset="/_next/image?url=%2Fimg%2Floginsideicon2.png&w=320&q=75 320w"
                style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`The <Login /> Component Should render the password field the user will see 1`] = `
<DocumentFragment>
  <section
    class="loginWrapper"
  >
    <div
      class="row"
    >
      <div
        class="loginSeparator1 col"
      >
        <a
          class="back__redirect nav-link"
          href="/"
        >
          <svg
            class="backButton"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
            />
          </svg>
           
          <span>
            Back
          </span>
        </a>
        <div
          style="max-width: 100%; width: 150px;"
        >
          <div
            style="position: relative; padding-bottom: 43.333333333333336%;"
          >
            <img
              alt="login"
              class="__lazy"
              data-src="/_next/image?url=%2Fimg%2Floginmatrix.png&w=320&q=75"
              data-srcset="/_next/image?url=%2Fimg%2Floginmatrix.png&w=320&q=75 320w"
              style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
          </div>
        </div>
        <div
          class="loginFormSubWrapper"
        >
          <div
            class="logo__section"
          >
            <div
              class="logo__title"
            >
              GetIdea
            </div>
            <div
              style="max-width: 100%; width: 35px;"
            >
              <div
                style="position: relative; padding-bottom: 108.57142857142857%;"
              >
                <img
                  alt="logo"
                  class="__lazy"
                  data-src="/_next/image?url=%2Fimg%2Flogo.png&w=320&q=75"
                  data-srcset="/_next/image?url=%2Fimg%2Flogo.png&w=320&q=75 320w"
                  style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
                />
              </div>
            </div>
          </div>
          <p
            class="loginPara"
          >
            Sign in to your account to continue
          </p>
          <form
            action="#"
            class="loginForm"
          >
            <div
              class="loginForm__seperator"
            >
              <label
                class="label"
                for="email"
              >
                Email
              </label>
              <input
                class="loginForm__input"
                id="email"
                name="email"
                type="text"
                value=""
              />
            </div>
            <div
              class="loginForm__seperator"
            >
              <label
                class="label"
                for="password"
              >
                Password
              </label>
              <input
                class="loginForm__input"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="loginForm__authRedirect"
            >
              <a
                class="loginForm__redirect nav-link"
                href="/forgot-password"
              >
                Forgot Password?
              </a>
              <a
                class="loginForm__redirect nav-link"
                href="/register"
              >
                Donâ€™t have an account 
                <span>
                  Register
                </span>
                ?
              </a>
            </div>
            <div
              class="loginForm__seperator"
            >
              <button
                class="loginForm__submit"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
        <div
          style="max-width: 100%; width: 647px;"
        >
          <div
            style="position: relative; padding-bottom: 21.020092735703248%;"
          >
            <img
              alt="login squares"
              class="loginSeparator1Img __lazy"
              data-src="/_next/image?url=%2Fimg%2Floginranbows.png&w=768&q=75"
              data-srcset="/_next/image?url=%2Fimg%2Floginranbows.png&w=320&q=75 320w, /_next/image?url=%2Fimg%2Floginranbows.png&w=420&q=75 420w, /_next/image?url=%2Fimg%2Floginranbows.png&w=768&q=75 768w"
              style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
          </div>
        </div>
      </div>
      <div
        class="loginSeparator2 col"
      >
        <div
          class="loginSideWrapper"
        >
          <div
            style="max-width: 100%; width: 50px;"
          >
            <div
              style="position: relative; padding-bottom: 90%;"
            >
              <img
                alt="login icon"
                class="__lazy"
                data-src="/_next/image?url=%2Fimg%2Floginsideicon.png&w=320&q=75"
                data-srcset="/_next/image?url=%2Fimg%2Floginsideicon.png&w=320&q=75 320w"
                style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
            </div>
          </div>
          <p
            class="loginSidePara"
          >
            Our support team are always ready 24/7 to help you start and grow your business
          </p>
          <div
            style="max-width: 100%; width: 50px;"
          >
            <div
              style="position: relative; padding-bottom: 50%;"
            >
              <img
                alt="login icon"
                class="__lazy"
                data-src="/_next/image?url=%2Fimg%2Floginsideicon2.png&w=320&q=75"
                data-srcset="/_next/image?url=%2Fimg%2Floginsideicon2.png&w=320&q=75 320w"
                style="visibility: hidden; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
